### PR TITLE
Resolves camel-14728. master branch; Rearrange website source to better use Antora capabilities and reduce…

### PR DESCRIPTION
**Pull request by djencks**
… duplicate pages.

### CODE REVIEW

File: core/camel-core-engine/src/main/docs/modules/eips/pages/hystrixConfiguration-eip.adoc [hunk 1]
```diff
@@ -10,34 +10,34 @@ The Hystrix Configuration EIP supports 31 options which are listed below:
...
+| *circuitBreakerForceClosed* | If true the HystrixCircuitBreaker#allowRequest() will always return true to allow requests regardless of the error percentage from HystrixCommandMetrics.getHealthCounts(). The circuitBreakerForceOpen() property takes precedence so if it set to true this property does nothing. | false | String
+| *circuitBreakerForceOpen* | If true the HystrixCircuitBreaker.allowRequest() will always return false, causing the circuit to be open (tripped) and reject all requests. This property takes precedence over circuitBreakerForceClosed(); | false | String
+| *circuitBreakerRequestVolume Threshold* | Minimum number of requests in the metricsRollingStatisticalWindowInMilliseconds() that must exist before the HystrixCircuitBreaker will trip. If below this number the circuit will not trip regardless of error percentage. | 20 | String
+| *circuitBreakerSleepWindowIn Milliseconds* | The time in milliseconds after a HystrixCircuitBreaker trips open that it should wait before trying requests again. | 5000 | String
```
Comment (davsclaus) [inline core/camel-core-engine/src/main/docs/modules/eips/pages/hystrixConfiguration-eip.adoc]: There is something in your change that is causing all the types of these EIPs to be wrong, eg the Integer -> String. And so on.

Comment (djencks) [reply core/camel-core-engine/src/main/docs/modules/eips/pages/hystrixConfiguration-eip.adoc]: Perhaps you could suggest what might be causing this: checking the (generated) json source for one of these the type is being copied from json to adoc accurately:
```
    "circuitBreakerEnabled": { "kind": "attribute", "displayName": "Circuit Breaker Enabled", "required": false, "type": "string", "javaType": "java.lang.String", "deprecated": false, "secret": false, "defaultValue": "true", "description": "Whether to use a HystrixCircuitBreaker or not. If false no circuit-breaker logic will be used and all requests permitted. This is similar in effect to circuitBreakerForceClosed() except that continues tracking metrics and knowing whether it should be open\/closed, this property results in not even instantiating a circuit-breaker." },
```

AFAICT I didn't touch any part of the code or build that could possibly affect the json generation.

**Comment by djencks** [comment]
I checked out master branch locally (from when I did the rebase yesterday for this) and built core and see the same change in type in circuitBreakerEnabled.  So I don't think I caused these changes.

**Comment by oscerd** [comment]
It seems to be related to CAMEL-13575

**Comment by oscerd** [comment]
This commit https://github.com/apache/camel/commit/a6deea370d6a2c3c64c8f2df4f02696ba2c12773

**Comment by oscerd** [comment]
So I think they were not updated before This PR

**Comment by djencks** [comment]
Yes, previous to my changes none of the eip docs were being updated from the json:
```
     private void executeEips() throws MojoExecutionException {
-        // only run if in camel-core
+        // only run if in camel-core-engine
         String currentDir = Paths.get(".").normalize().toAbsolutePath().toString();
-        if (!currentDir.endsWith("camel-core")) {
+        if (!currentDir.endsWith("camel-core-engine")) {
             return;
         }
```

**Comment by oscerd** [comment]
Better, we were wrong then :-)
@davsclaus, I think once this is rebased on top of master, we can merge this.

**Comment by davsclaus** [comment]
Ah yeah we made the EIP use string types so they can be configured via property placeholders. So yeah this is fine.

**Comment by oscerd** [comment]
Btw I saw some of the eips are no more in the user manual like https://github.com/apache/camel/pull/3646/files#diff-e16200cc422d4abef97c09facc9f2847

is this wanted? Am I missing something?

**Comment by djencks** [comment]
Do you mean `languages` rather than `eips`?  IIRC some of the language pages were duplicated in the user manual and in components.  I moved them all to components.  I don't think having two locations for the same documentation page is usually a good idea.

**Comment by oscerd** [comment]
No, I was talking about stream-config-eip, because that's the one I noticed, but there are more. I don't think it is problem if we are grouping them somewhere else.

**Comment by djencks** [comment]
If I understand what you mean, this is an effect of one of the main ideas in this change.  Instead of copying all the eip pages into "user-manual" they are included directly from their original location in core/camel-core-engine.  That's why I changed the directory structure in core/camel-core-engine/src/main/docs to match Antora structure.

**Comment by davsclaus** [comment]
AFAIR then the file names was named -eip if they were partly auto-generated, eg they are a model in the source code. And then we had other conceptual EIPs which is just some theory text, like messaging channel (no -eip in the name)

**Comment by oscerd** [comment]
Yes, that's the point, I was missing this "little" details.

**Comment by djencks** [comment]
It looks like these might be other eip pages:
```
    Occurrences of 'http://www.enterpriseintegrationpatterns.com' in directory /Users/david/projects/camel/camel/docs/user-manual/modules/ROOT/pages with mask '*.adoc'
Found Occurrences  (22 usages found)
    Unclassified occurrence  (22 usages found)
        camel  (22 usages found)
            /Users/david/projects/camel/camel/docs/user-manual/modules/ROOT/pages  (22 usages found)
                books.adoc  (3 usages found)
                    23 * http://www.enterpriseintegrationpatterns.com/[Enterprise Integration
                    219 http://www.enterpriseintegrationpatterns.com/[Enterprise Integration
                    223 http://www.enterpriseintegrationpatterns.com/[image:books/eip_book_cover.jpg[image]]
                content-enricher.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/DataEnricher.html[Content
                correlation-identifier.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/CorrelationIdentifier.html[Correlation
                dead-letter-channel.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/DeadLetterChannel.html[Dead
                event-message.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/EventMessage.html[Event
                guaranteed-delivery.adoc  (2 usages found)
                    5 http://www.enterpriseintegrationpatterns.com/GuaranteedMessaging.html[Guaranteed
                    25 of http://www.enterpriseintegrationpatterns.com/GuaranteedMessaging.html[Guaranteed
                message-bus.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/PointToPointChannel.html[Message
                message-channel.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/MessageChannel.html[Message
                message-endpoint.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/MessageEndpoint.html[Message
                message-router.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/MessageRouter.html[Message
                message-translator.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/MessageTranslator.html[Message
                message.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/Message.html[Message] from
                point-to-point-channel.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/PointToPointChannel.html[Point
                polling-consumer.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/PollingConsumer.html[Polling
                publish-subscribe-channel.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/PublishSubscribeChannel.html[Publish
                request-reply.adoc  (1 usage found)
                    6 http://www.enterpriseintegrationpatterns.com/RequestReply.html[Request
                return-address.adoc  (1 usage found)
                    4 http://www.enterpriseintegrationpatterns.com/ReturnAddress.html[Return
                transactional-client.adoc  (1 usage found)
                    5 http://www.enterpriseintegrationpatterns.com/TransactionalClient.html[Transactional
                walk-through-another-example.adoc  (1 usage found)
                    13 particular we focus on http://www.enterpriseintegrationpatterns.com/PipesAndFilters.html[Pipes and Filters] -
```

(perhaps not the first and last).

I'd suggest moving these to core/core-camel-engine/ to be with the generated eip pages. Otherwise they could be moved to an eip module in user-manual.  I'd rather do this in a subsequent issue/PR, WDYT?

**Comment by oscerd** [comment]
I think it's ok. +1

**Comment by djencks** [comment]
I rebased on master (what a lot of changes since yesterday!) and force-pushed.  I'm not familiar with the camel workflow.  Will one of you merge this when you're satisfied or should I (although I'm an initial committer on camel, I don't think I've ever committed before).  I'd be happier with one of you merging.

**Comment by oscerd** [comment]
Thanks a lot, I'm going to start the job on the camel-website and see how the PR goes

**Pull request merged**